File: tests/XMP/global-view/shadow-reflect/C/issue41.c
#include <stdio.h>

typedef struct test {
  double v;
} test_t;

#pragma xmp template t(0:9,0:9)
#pragma xmp nodes p(2,2)
#pragma xmp distribute t(block, block) onto p
test_t a[10][10];
#pragma xmp align a[i][j] with t(j,i)
#pragma xmp shadow a[1][1]

int main()
{
#pragma xmp reflect (a)  async(1)
#pragma xmp wait_async(1)

#pragma xmp task on p(1,1)
  {
    printf("PASS\n");
  }
  
  return 0;
}

File: tests/XMP/global-view/shadow-reflect/F/issue41.f90
program main

  type, bind(c) :: test
     real v
     real(8) w, x
  end type test

  !$xmp nodes p(2,2)

  !$xmp template t(10,10)

  !$xmp distribute t(block, block) onto p
  type(test) a(10,10)
  !$xmp align a(i,j) with t(i,j)
  !$xmp shadow a(1,1)

  !$xmp reflect (a) async(1)
  !$xmp wait_async(1)

!$xmp task on p(1,1)
  write(*,*) "PASS"
!$xmp end task

end program main
